File: SDKs/microsoft.com/cognitiveservices-Training/3.1/typescript/src/sdk/models/operations/suggesttagsandregions.ts
import { SpeakeasyMetadata, SpeakeasyBase } from "../../../internal/utils";
import * as shared from "../shared";



export class SuggestTagsAndRegionsPathParams extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "pathParam, style=simple;explode=false;name=projectId" })
  projectId: string;
}


export class SuggestTagsAndRegionsQueryParams extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "queryParam, style=form;explode=false;name=imageIds" })
  imageIds: string[];

  @SpeakeasyMetadata({ data: "queryParam, style=form;explode=true;name=iterationId" })
  iterationId: string;
}


export class SuggestTagsAndRegionsHeaders extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "header, style=simple;explode=false;name=Training-Key" })
  trainingKey: string;
}


export class SuggestTagsAndRegionsRequest extends SpeakeasyBase {
  @SpeakeasyMetadata()
  pathParams: SuggestTagsAndRegionsPathParams;

  @SpeakeasyMetadata()
  queryParams: SuggestTagsAndRegionsQueryParams;

  @SpeakeasyMetadata()
  headers: SuggestTagsAndRegionsHeaders;
}


export class SuggestTagsAndRegionsResponse extends SpeakeasyBase {
  @SpeakeasyMetadata()
  body?: Uint8Array;

  @SpeakeasyMetadata()
  contentType: string;

  @SpeakeasyMetadata()
  customVisionError?: shared.CustomVisionError;

  @SpeakeasyMetadata()
  statusCode: number;

  @SpeakeasyMetadata({ elemType: shared.SuggestedTagAndRegion })
  suggestedTagAndRegions?: shared.SuggestedTagAndRegion[];
}

File: SDKs/microsoft.com/cognitiveservices-Training/3.1/python/src/sdk/models/operations/suggesttagsandregions.py
import dataclasses
from typing import Optional
from ..shared import customvisionerror as shared_customvisionerror
from ..shared import suggestedtagandregion as shared_suggestedtagandregion


@dataclasses.dataclass
class SuggestTagsAndRegionsPathParams:
    project_id: str = dataclasses.field(metadata={'path_param': { 'field_name': 'projectId', 'style': 'simple', 'explode': False }})
    

@dataclasses.dataclass
class SuggestTagsAndRegionsQueryParams:
    image_ids: list[str] = dataclasses.field(metadata={'query_param': { 'field_name': 'imageIds', 'style': 'form', 'explode': False }})
    iteration_id: str = dataclasses.field(metadata={'query_param': { 'field_name': 'iterationId', 'style': 'form', 'explode': True }})
    

@dataclasses.dataclass
class SuggestTagsAndRegionsHeaders:
    training_key: str = dataclasses.field(metadata={'header': { 'field_name': 'Training-Key', 'style': 'simple', 'explode': False }})
    

@dataclasses.dataclass
class SuggestTagsAndRegionsRequest:
    headers: SuggestTagsAndRegionsHeaders = dataclasses.field()
    path_params: SuggestTagsAndRegionsPathParams = dataclasses.field()
    query_params: SuggestTagsAndRegionsQueryParams = dataclasses.field()
    

@dataclasses.dataclass
class SuggestTagsAndRegionsResponse:
    content_type: str = dataclasses.field()
    status_code: int = dataclasses.field()
    body: Optional[bytes] = dataclasses.field(default=None)
    custom_vision_error: Optional[shared_customvisionerror.CustomVisionError] = dataclasses.field(default=None)
    suggested_tag_and_regions: Optional[list[shared_suggestedtagandregion.SuggestedTagAndRegion]] = dataclasses.field(default=None)
    
